src/components/autoFocuser.js

```diff
@@ -0,0 +1,96 @@
+define(["focusManager", "layoutManager"], function (focusManager, layoutManager) {
+    "use strict";
+
+    /**
+     * Previously selected element.
+     */
+    var activeElement;
+
+    /**
+     * Returns true if AutoFocuser is enabled.
+     */
+    function isEnabled() {
+        return layoutManager.tv;
+    };
+
+    /**
+     * Start AutoFocuser
+     */
+    function enable() {
+        if (!isEnabled()) {
+            return;
+        }
+
+        window.addEventListener("focusin", function (e) {
+            activeElement = e.target;
+        });
+
+        console.log("AutoFocuser enabled");
+    }
+
+    /**
+     * Create an array from some source.
+     */
+    var arrayFrom = Array.prototype.from || function (src) {
+        return Array.prototype.slice.call(src);
+    }
+
+    /**
+     * Set focus on a suitable element, taking into account the previously selected.
+     */
+    function autoFocus(container) {
+        if (!isEnabled()) {
+            return;
+        }
+
+        container = container || document.body;
+
+        var candidates = [];
+
+        if (activeElement) {
+            // These elements are recreated
+            if (activeElement.classList.contains("btnPreviousPage")) {
```

Comment:
can we do this as a loop over an array of all potential candidates handled similar here?

Reply:
I planned:
if `btnPreviousPage` was focused then it should be focused next, but if it is disabled, then `btnNextPage` should. And vice versa. Then user stays on paging buttons for as long as necessary.

To make this logic as loop :thinking: :
```
var focusRules = [
    {
        prevCss: ".btnPreviousPage",
        nextCss: [".btnNextPage", ".btnPreviousPage"]
    },
    {
        prevCss: ".btnNextPage",
        nextCss: [".btnPreviousPage", ".btnNextPage"]
    },
    {
        prevCss: ".btnSelectView"
        // Suppose nextCss: [prevCss]
    },

    // Here we must check activeElement as it is
    {}, // unobvious HACK :)

    // If no prevCss, then this could be perceived as "no activeElement"
    {
        nextCss: [".btnResume", ".btnPlay"]
    }
];
```

Reply:
okay, I see your point... I was thinking too Pythonic here, js arrays and for-s might be too strict to implement

<details><summary>Python illustration</summary>

```python
for prevCss, nextCss in ((".btnPreviousPage", (".btnNextPage", ".btnPreviousPage")),
                         (".btnNextPage", (".btnPreviousPage", ".btnNextPage")),
                         (".btnSelectView", (".btnSelectView",))):
    if activeElement.classList.contains(prevCss):
        for nextItem in nextCss:
            candidates.append(container.querySelector(nextItem))
```

</details>